crates/api_common/src/request.rs

```diff
@@ -200,50 +205,59 @@ pub async fn fetch_site_data(
       // Warning, this may ignore SSL errors
       let metadata_option = fetch_site_metadata(client, url).await.ok();
       if !include_image {
-        return (metadata_option, None);
+        (metadata_option, None)
+      } else {
+        let thumbnail_url =
+          match fetch_pictrs_hash(client, &metadata_option, settings, url).await {
+            Ok(pictrs_hash) => Url::parse(&format!(
```

Comment:
It'd also probably be cleaner to 

- move the `Url::parse` below into `fetch_pictrs_hash`.
- Change its name to something like `fetch_pictrs_url_from_site_metadata`, or whatever you like
- Have it return a `Url` rather than a string.